jupyterlab_gcedetails/src/components/synchronized_sliders.tsx

```diff
@@ -0,0 +1,246 @@
+    });
+
+    this.props.onCpuChange(newCpu);
+  }
+
+  private onMemoryChange(newMemory: number) {
+    this.setState({
+      cpuValue: this.cpuFromMemory(newMemory),
+      cpuInputDisplayValue: String(this.cpuFromMemory(newMemory)),
+      memoryValue: newMemory,
+      memoryInputDisplayValue: String(newMemory),
+    });
+
+    this.props.onMemoryChange(newMemory);
+  }
+
+  render() {
+    const {
+      cpuInputDisplayValue,
+      memoryInputDisplayValue,
+      cpuValue,
+      memoryValue,
+    } = this.state;
+
+    const handleCpuSliderChange = (event: any, newValue: number) => {
+      this.onCpuChange(newValue);
+    };
+
+    const handleMemorySliderChange = (event: any, newValue: number) => {
+      this.onMemoryChange(newValue);
+    };
+
+    const handleCpuInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      this.setState({
+        cpuInputDisplayValue: event.target.value,
+      });
+    };
+
+    const handleCpuInputBlur = () => {
+      const newValue = this.closest(
+        Number(cpuInputDisplayValue),
+        this.cpuValues
+      );
+      this.onCpuChange(newValue);
+    };
+
+    const handleMemoryInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      this.setState({
+        memoryInputDisplayValue: event.target.value,
+      });
+    };
+
+    const handleMemoryInputBlur = () => {
+      const newValue = this.closest(
+        Number(memoryInputDisplayValue),
+        this.memoryValues
+      );
+      this.onMemoryChange(newValue);
+    };
+
+    return (
+      <div>
+        <InputSlider
+          possibleValues={this.cpuValues}
+          label="Cores"
+          value={cpuValue}
+          inputDisplayValue={cpuInputDisplayValue}
+          handleSliderChange={handleCpuSliderChange}
+          handleInputChange={handleCpuInputChange}
+          handleInputBlur={handleCpuInputBlur}
+          helperText={`${this.cpuValues[0]}  - ${
+            this.cpuValues[this.cpuValues.length - 1]
+          } vCPUs`}
+        />
+        <InputSlider
+          possibleValues={this.memoryValues}
+          label="Memory"
+          value={memoryValue}
+          inputDisplayValue={memoryInputDisplayValue}
+          handleSliderChange={handleMemorySliderChange}
+          handleInputChange={handleMemoryInputChange}
+          handleInputBlur={handleMemoryInputBlur}
+          helperText={`${this.memoryValues[0]} - ${
+            this.memoryValues[this.memoryValues.length - 1]
+          } GB`}
+        />
+      </div>
+    );
+  }
+}
+
+interface InputSliderProps {
+  value: number;
+  possibleValues: number[];
+  label?: string;
+  handleSliderChange: (event: any, newValue: number) => void;
+
+  inputDisplayValue: string;
+  handleInputChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleInputBlur: () => void;
+  helperText?: string;
+}
+
+function InputSlider(props: InputSliderProps) {
```

Comment:
I could see this eventually being useful as a shared component, but I don't think you need to do that now. Still, it might be worth lifting it out to its own file.